frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -561,9 +576,69 @@ function ReaderHighlight:_resetHoldTimer(clear)
     end
 end
 
+function ReaderHighlight:getRegionalZoomCenter(pos)
+    if not self.dimen then
+        self.dimen = self.ui.dimen
+    end
+    local page_size = self.ui.document:getNativePageDimensions(pos.page)
+    local margin = self.ui.document.configurable.page_margin * Screen:getDPI()
+    local zoom = 2*self.dimen.w / page_size.w
+    zoom = zoom/(1 + 3*margin/zoom/page_size.w)
+    local x_ratio = (pos.x/ page_size.w) -- x center ratio
+    local y_ratio = (pos.y / page_size.h) -- y center ratio
+       
+    return zoom, x_ratio, y_ratio
+    
+end
+
+
 function ReaderHighlight:onHold(arg, ges)
     -- disable hold gesture if highlighting is disabled
-    if self.view.highlight.disabled then return false end
+    if self.view.highlight.disabled then
+        if self.view.highlight.zoom then
+            --zoom the hold area
+            self:clear() -- clear previous highlight (delayed clear may not have done it yet)
+            self.hold_ges_pos = ges.pos -- remember hold original gesture position
+            self.hold_pos = self.view:screenToPageTransform(ges.pos)
+            logger.dbg("hold position in page ", self.hold_pos)
+            if not self.hold_pos then
+                logger.dbg("not inside page area")
+                return false
+            end
+            
+            -- Screen:shot("/home/m/1.png")
+            logger.dbg("hold on image")
+
+            local getImage = require("document/koptinterface")
+            local Image = getImage:getNativePageImage(self.ui.document, self.hold_pos.page)
+            if Image then
+                logger.dbg("we got image")
+                local zoom, x_ratio, y_ratio
+                zoom, x_ratio, y_ratio = self:getRegionalZoomCenter(self.hold_pos)
+                logger.dbg("zoom:", zoom)
+                logger.dbg("xpos:", x_ratio)
+                logger.dbg("ypos:", y_ratio)
+                
+                --call magnifier widget
+                local Magnifier = require("ui/widget/magnifier")
+                local magnifier = Magnifier:new{
+                image = Image,
+                timeout = 5,
+                -- set zoom to dubble now - setting variable to be add
+                zoom = zoom * 2 ,
+                x_ratio = x_ratio,
+                y_ratio = y_ratio
+
+                }
+            UIManager:show(magnifier)
+            end
+            
+            return false
```

Comment:
You handled something: you return true, so the event does not propagate to other lower widgets.
When you don't and want to let it propagate as some lower widget might handle it: you return false.

Reply:
done